test/general-use.js

```diff
@@ -33,17 +33,22 @@ module.exports = {
     english: {
 
         // english (appropriate mode) (sorted)
+        "Jack": "anga;quesse:a,tilde-below",
+        "happy": "hyarmen;parma:a,tilde-below;long-carrier:y",
+        "style": "silme;tinco;lambe:y,i-below",
+        "yellow": "anna;lambe:e,tilde-below;vala:o",
+        "phone": "formenparma;numen:o,i-below",
         "cake": "quesse;quesse:a,i-below",
         "cakes": "quesse;quesse:a;silme-nuquerna:e",
         "cats.": "quesse;tinco:a,s-final;full-stop", // regression
-        "green": "ungwe;romen;long-carrier:e;numen",
+        "green": "ungwe;romen;short-carrier:e;numen:e",
         "hobbits": "hyarmen;umbar:o,tilde-below;tinco:i,s-final",
         "hobbits'": "hyarmen;umbar:o,tilde-below;tinco:i,s-inverse",
         "hobbits''": "hyarmen;umbar:o,tilde-below;tinco:i,s-extended",
         "hobbits'''": "hyarmen;umbar:o,tilde-below;tinco:i,s-flourish",
-        "there": "thule;romen:e,i-below",
+        "there": "thule;ore:e,i-below",
         "these": "thule;silme-nuquerna:e;short-carrier:i-below",
-        "these'": "thule;silme-nuquerna:e;short-carrier:e",
+        "these'": "thule;silme-nuquerna:e;short-carrier:e", // invalid input
```

Comment:
Let’s add tests for words with a final "e" that would be the exceptions to the rule like "see" and "naïve". In these cases, we will need to use a notation to allow the user to override the default behavior, and maybe even hard-code as many known exceptions as possible.

I’ll note that pattern matching vowel-consonant-vowel to distinguish the silent E case from other voiced final E cases may require extraordinary programming.

Anything valid should be expressible, even if it requires manual intervention to express. That’s the current function of the apostrophe.

Reply:
Is there anyways to make classifications for the letters so it can tell "symbol from this list is vowel, and symbol from this list is consonant" ? Or is this what you mean by "extraordinary programming"? I don't know enough about this sort of thing to be able to tell if this would be too complex or not.

Reply:
Classifying vowels and consonants is easy, except for the cases like w and y where they’re ambiguous for the purposes of English rules.

Matching patterns of three phonemes requires considerable additional work because it involves back-tracking and also taking into account consonant and vowel clusters (not a one character to one phoneme relation). Far easier to pass the burden of distinguishing silent and voiced final e to the user, just assuming e is unvoiced and ë or e` are voiced.

As in the case of naïve, to infer that the final e is voiced from the diaeresis over the ï, we’d have to thread a hint forward, over the following consonant and into the code that matches final e. That is not so hard, but threading the additional state requires altering most of the function calls in the transcriber.